src/test/components/elements/EventBookingForm.test.tsx

```diff
@@ -0,0 +1,193 @@
+import { API_PATH, augmentEvent } from "../../../app/services";
+import { AdditionalInformation, AugmentedEvent } from "../../../IsaacAppTypes";
+import { Immutable } from "immer";
+import { fireEvent, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { rest } from "msw";
+import * as actions from "../../../app/state/actions";
+
+const requestEmailVerificationSpy = jest.spyOn(actions, "requestEmailVerification");
+const updateAdditionalInformation = jest.fn();
+
+describe("EventBookingForm", () => {
+  const setupTest = ({
+    role,
+    event,
+    user,
+    additionalInformation,
+  }: {
+    role: UserRole;
+    event?: AugmentedEvent;
+    user: typeof mockUser;
+    additionalInformation?: AdditionalInformation;
+  }) => {
+    const mockUserSummary: Immutable<UserSummaryWithEmailAddressDTO> = {
+      givenName: user.givenName,
+      familyName: user.familyName,
+      role: role,
+      emailVerificationStatus: user.emailVerificationStatus as EmailVerificationStatus,
+      registeredContexts: user.registeredContexts as UserContext[],
+      id: user.id,
+      teacherPending: user.teacherPending,
+    };
+
+    renderTestEnvironment({
+      role: role,
+      PageComponent: EventBookingForm,
+      componentProps: {
+        event: event ? event : augmentEvent(mockEvent),
+        targetUser: mockUserSummary,
+        additionalInformation: additionalInformation ? additionalInformation : {},
+        updateAdditionalInformation: updateAdditionalInformation,
+      },
+      initialRouteEntries: [`/events/${mockEvent.id}`],
+      extraEndpoints: [
+        rest.post(API_PATH + `/users/verifyemail`, (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json({}));
+        }),
+      ],
+    });
+  };
+
+  it("component renders", () => {
+    setupTest({ role: "STUDENT", user: mockUser });
+    const accountInformation = screen.getByTestId("booking-account-info");
+    const eventBookingDetails = screen.getByTestId("event-booking-details");
+    expect(accountInformation).toBeInTheDocument();
+    expect(eventBookingDetails).toBeInTheDocument();
+  });
+
+  it("has first name, last name , email address, context and school inputs, all disabled", async () => {
+    setupTest({ role: "STUDENT", user: mockUser });
+    await screen.findByDisplayValue(mockUser.givenName!);
+    const inputFields = [
+      "First name",
+      "Last name",
+      "Email address",
+      "Stage",
+      "Exam board",
+      "My current school or college",
+    ];
+    inputFields.forEach((input) => {
+      const inputField = screen.getByLabelText(input);
+      expect(inputField).toBeInTheDocument();
+      expect(inputField).toBeDisabled();
+    });
+  });
+
+  it("if email address is unverified, warning message is displayed", async () => {
+    const user = { ...mockUser, emailVerificationStatus: "NOT_VERIFIED" as EmailVerificationStatus };
```

Comment:
Did you have the chance to investigate const assertions? Is this preferred to `as const`? In other words: in which situation should you use one or the other?

Reply:
I did read up on it a bit, and both uses would work here I believe, but `EmailVerificationStatus` is more explicit - there's no additional benefit to using `const` as far as I can tell.  Given that we know what type it actually is, that makes more sense to me - probably just a stylistic decision at this stage!